run_linux.go

```diff
@@ -2829,25 +2840,27 @@ func (b *Builder) getSSHMount(tokens []string, count int, sshsources map[string]
 		return nil, nil, err
 	}
 	// Start ssh server, and get the host sock we're mounting in the container
-	hostSock, err := fwdAgent.Serve(processLabel)
+	hostSock, err := fwdAgent.Serve(b.ProcessLabel)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	if err := label.Relabel(filepath.Dir(hostSock), mountlabel, false); err != nil {
+	if err := label.Relabel(filepath.Dir(hostSock), b.MountLabel, false); err != nil {
 		if shutdownErr := fwdAgent.Shutdown(); shutdownErr != nil {
 			b.Logger.Errorf("error shutting down agent: %v", shutdownErr)
 		}
 		return nil, nil, err
 	}
-	if err := label.Relabel(hostSock, mountlabel, false); err != nil {
+	if err := label.Relabel(hostSock, b.MountLabel, false); err != nil {
 		if shutdownErr := fwdAgent.Shutdown(); shutdownErr != nil {
 			b.Logger.Errorf("error shutting down agent: %v", shutdownErr)
 		}
 		return nil, nil, err
 	}
-
-	hostUID, hostGID, err := util.GetHostIDs(uidmap, gidmap, uid, gid)
+	hostUID, hostGID, err := util.GetHostIDs(idMaps.uidmap, idMaps.gidmap, uid, gid)
+	if err != nil {
```

Comment:
I think this `err !=nil` block is not needed since we already handle `err` in the line below and perform shutdown of ssh-agent if it was created, following line will override the old logic.